### PR TITLE
Coding Project

### PR DESCRIPTION
**Pull request by CurtiePi**
Hello I am done with the basic assignment, thought I would like to add some more comments and set up procedures. I have created a dev branch to implement oauth2 features, which is what I am currently trying to do. If I cannot fully implement it in the next 20 hours, I will merge what I have with master and turn that in. 

Feedback is greatly appreciated

### CODE REVIEW

File: routes/securityRoutes.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+            })
+            .catch((err) => {
+                res.status(500).send('Houston we have a problem!');
+            });
+    });
+
+    router.post('/login', passport.authenticate('local', {
+        failureRedirect:'/',
+        successRedirect:'/user/list'
+    }), (req, res) => {
```
Comment (woodpig07) [inline routes/securityRoutes.js]: Hi Curtis,
When I was testing the login function, the page redirected me  to route `/user/list` with a blank user info page.
I could see the session cookie was successfully sent to browser, but the page did not render the registered user data. Would you check why is that happening?

**Comment by CurtiePi** [comment]
Sure thing

**Comment by CurtiePi** [comment]
Hello Kate,
Can you provide me with the steps you took to receive that error so that I
can recreate.

I'm going to assume you started the M I ngoDB.

Did you register a new user or did you use the test user (dolemite)?

I'm trying to recreate your error on this

Curtis

**Comment by CurtiePi** [comment]
Hello again Kate,

It sounds like you're telling me that when you login in, you log in
successfully, and you are redirected to the /user/list page, however there
are no users listed.

So are you getting an actual exception or just an empty list of users?

If you are getting an empty list of users then I am wondering if you ran
the command *node ./libs/dataGen.js* in the root directory. This function
should populate the database with person documents that should be seen on
the /user/list page.

If you received an actual exception then please send me the steps to
recreate.

thank you
curtis

**Comment by CurtiePi** [comment]
Hello Kate,

I'm sending a video of how I prepare the data and login so that you can see
what to expect and compare.

I will say one problem I see is a screen message telling the user (you in
this case) that the result list of the distance search is active. I will
correct that while I look for your aforementioned problem.

cheers
curtis
 start_login.mov
<https://drive.google.com/file/d/1PV5BD_mxF0AY6vbOmJeWHPqrqeKsIrnQ/view?usp=drive_web>

**Comment by CurtiePi** [comment]
Hello Kate,

so far I have not been able to recreate your issue, so I'll wait for more
feedback. However, I found a minor issue on my own and I have
corrected it. It's nothing that should stop you from using the site, but if
you pull the latest (as of 9:10pm Thursday) you should  have the most
recent codebase.

looking forward to hearing from you

curtis

**Comment by CurtiePi** [comment]
Hello again Kate,

If you think weChatting would help resolve the situation my weChat id is
*CurtiePie.*

aloha

curtis

**Comment by woodpig07** [comment]
Hi Curtis,
Thank you for your prompt reply response.
It's better to keep our conversation here 😄 

I did run script to populate data to mongodb before bringing up the app.
Here are some screen casts
[Screen Recording 2019-12-19 at 10.42.36 PM.mov.zip](https://github.com/Wiredcraft/test-backend/files/3984126/Screen.Recording.2019-12-19.at.10.42.36.PM.mov.zip)
[Screen Recording 2019-12-19 at 10.39.22 PM.mov.zip](https://github.com/Wiredcraft/test-backend/files/3984128/Screen.Recording.2019-12-19.at.10.39.22.PM.mov.zip)

**Comment by CurtiePi** [comment]
Hey Kate,

sorry you're up so late. Me I'm always up late.

Thanks for the screen capture; I know what the problem is and I'll fix it
now. I left my test code in the signup route.

I'll fix, test and push. I'll send you a message after I have pushed to the
master branch.

Sorry for the inconvenience.

curtis

**Comment by CurtiePi** [comment]
Hello Kate,
So I have pushed the fix to master and it's ready for you to have a look.

So the issue was, for security I was not putting the user into the session
on sign up. Yes it's a pain, but after you sign up you must log in.

So now you will/should not see the user information on sign-up, but you
will be redirected back to the login page. Once you log in then your user
info will be put into the session and you will see the the user list.

**Comment by woodpig07** [comment]
Hi Curtis,

I finally figured out why I saw an empty user list page after successfully login.

The script to insert Person data `node ./libs/dataGen.js` is connecting to database `wired_backend_dev`

https://github.com/Wiredcraft/test-backend/blob/6a2a6f317ed1aa91c8a98f05c15b65214b63c61f/libs/dataGen.js#L8-L10

Whereas in the environment config file, the default database is `wired_db`

https://github.com/Wiredcraft/test-backend/blob/6a2a6f317ed1aa91c8a98f05c15b65214b63c61f/config/config.js#L5-L15

Also for the unit tests, when I tried to run `npm test`, it looks a lot of test failed because of mongo database connection issue. I believe that's because it has no database connection initialisation before running each test.

**Comment by CurtiePi** [comment]
Hey Kate,

That's odd as the documentation says the environment variable should be
*wired_backend_dev* and that *.env* file is git ignored.

So as for the tests, in each test suite I have a line of code *const app =
require('../btapp');* which initializes the database connection, however I
can rewrite the test to explicitly initialize the connections for each test.

Give me a few minutes.

curtis

**Comment by CurtiePi** [comment]
Hello Kate,

The tests have been modified, you can pull and retest.

What was done?


   - I created a* config/dbservice.js* file to initialize the database
   connection
   - In *test/basic_user_test.js *and* test/dac_test.js *I removed the
   require to the app file (*btapp.js*)
   - In the two aforementioned tests I required* config/dbservice.js*
   - In the two aforementioned tests I use Mocha's* beforeEach hook *to
   create the connection before each test is run
   - in *test/route_api_test.js *I remove all people from the database

So, with the way the tests are written now you can start with an empty
database (previously there was a testDataGen.js file which I will remove
that loaded the test database with initial data). So now you should be able
to run the test from start to finish. I'm keeping my fingers crossed,
because superstition.

curtis

**Comment by woodpig07** [comment]
Thanks for the quick updates. We'll continue the review.

**Comment by CurtiePi** [comment]
Thank you.

**Pull request closed without merging**